Refresh.Interfaces.APIv3/Endpoints/CategoryApiEndpoints.cs

```diff
@@ -0,0 +1,128 @@
+    public ApiListResponse<ApiGameLevelResponse> GetLevels(RequestContext context, CategoryService categories, GameUser? user,
+        [DocSummary("The name of the category you'd like to retrieve levels from. " +
+                    "Make a request to /levels to see a list of available categories")]
+        string route, DataContext dataContext)
+    {
+        if (string.IsNullOrWhiteSpace(route))
+        {
+            return new ApiError("You didn't specify a route. " +
+                                "You probably meant to use the `/levels` endpoint and left a trailing slash in the URL.", NotFound);
+        }
+        
+        (int skip, int count) = context.GetPageData();
+
+        DatabaseList<GameLevel>? list = categories.LevelCategories
+            .FirstOrDefault(c => c.ApiRoute.StartsWith(route))?
+            .Fetch(context, skip, count, dataContext, new LevelFilterSettings(context, TokenGame.Website), user);
+
+        if (list == null) return ApiNotFoundError.Instance;
+
+        DatabaseList<ApiGameLevelResponse> levels = DatabaseListExtensions.FromOldList<ApiGameLevelResponse, GameLevel>(list, dataContext);
+        return levels;
+    }
+
+    [ApiV3Endpoint("users"), Authentication(false)]
+    [ClientCacheResponse(86400 / 2)] // cache for half a day
+    [DocSummary("Retrieves a list of categories you can use to search users. Returns an empty list if the instance doesn't allow showing online users.")]
+    [DocQueryParam("includePreviews", "If true, a single user will be added to each category representing a user from that category. False by default.")]
+    [DocError(typeof(ApiValidationError), "The boolean 'includePreviews' could not be parsed by the server.")]
+    public ApiListResponse<ApiUserCategoryResponse> GetUserCategories(RequestContext context, CategoryService categories,
+        DataContext dataContext, GameServerConfig config)
+    {
+        bool result = bool.TryParse(context.QueryString.Get("includePreviews") ?? "false", out bool includePreviews);
+        if (!result) return ApiValidationError.BooleanParseError;
+
+        if (!config.PermitShowingOnlineUsers) return new ApiListResponse<ApiUserCategoryResponse>([]);
+        IEnumerable<ApiUserCategoryResponse> resp;
+
+        if (includePreviews) resp = ApiUserCategoryResponse.FromOldList(categories.UserCategories, context, dataContext);
+        else resp = ApiUserCategoryResponse.FromOldList(categories.UserCategories, dataContext);
+
+        return new ApiListResponse<ApiUserCategoryResponse>(resp);
+    }
+
+    // This route can not be called "users/{route}", else Bunkum will route users/me requests to here aswell.
+    // Having a special case for the "me" route here would be hacky and introduce trouble if another endpoint with the route
+    // "users/something" (for example) were to ever be implemented in the future.
+    [ApiV3Endpoint("users/category/{route}"), Authentication(false)]
```

Comment:
Hmm... I personally think it would be okay to have this hack, actually. We do similar things in the Game API code.